Stripe/STPSource.m

```diff
@@ -214,34 +214,51 @@ - (NSString *)stripeObject {
     return @"source";
 }
 
-+ (NSArray *)requiredFields {
-    return @[@"id", @"livemode", @"status", @"type"];
-}
-
 + (instancetype)decodedObjectFromAPIResponse:(NSDictionary *)response {
-    NSDictionary *dict = [response stp_dictionaryByRemovingNullsValidatingRequiredFields:[self requiredFields]];
+    NSDictionary *dict = [response stp_dictionaryByRemovingNulls];
     if (!dict) {
         return nil;
     }
 
+    // required fields
+    NSString *stripeId = [dict stp_stringForKey:@"id"];
+    NSString *rawStatus = [dict stp_stringForKey:@"status"];
+    NSString *rawType = [dict stp_stringForKey:@"type"];
+    if (!stripeId || !rawStatus || !rawType || !dict[@"livemode"]) {
+        return nil;
+    }
+
     STPSource *source = [self new];
-    source.stripeID = dict[@"id"];
-    source.amount = dict[@"amount"];
-    source.clientSecret = dict[@"client_secret"];
-    source.created = [NSDate dateWithTimeIntervalSince1970:[dict[@"created"] doubleValue]];
-    source.currency = dict[@"currency"];
-    source.flow = [[self class] flowFromString:dict[@"flow"]];
-    source.livemode = [dict[@"livemode"] boolValue];
-    source.metadata = [dict[@"metadata"] stp_dictionaryByRemovingNonStrings];
-    source.owner = [STPSourceOwner decodedObjectFromAPIResponse:dict[@"owner"]];
-    source.receiver = [STPSourceReceiver decodedObjectFromAPIResponse:dict[@"receiver"]];
-    source.redirect = [STPSourceRedirect decodedObjectFromAPIResponse:dict[@"redirect"]];
-    source.status = [[self class] statusFromString:dict[@"status"]];
-    NSString *typeString = dict[@"type"];
-    source.type = [[self class] typeFromString:typeString];
-    source.usage = [[self class] usageFromString:dict[@"usage"]];
-    source.verification = [STPSourceVerification decodedObjectFromAPIResponse:dict[@"verification"]];
-    source.details = dict[typeString];
+    source.stripeID = stripeId;
+    source.amount = [dict stp_numberForKey:@"amount"];
+    source.clientSecret = [dict stp_stringForKey:@"client_secret"];
+    source.created = [dict stp_dateForKey:@"created"];
+    source.currency = [dict stp_stringForKey:@"currency"];
+    NSString *rawFlow = [dict stp_stringForKey:@"flow"];
+    source.flow = [[self class] flowFromString:rawFlow];
+    source.livemode = [dict stp_boolForKey:@"livemode" or:NO];
```

Comment:
In `STPEphemeralKey` and `STPToken`, you coerce un-parseable `livemode` values to `YES`. Any reason for the discrepancy here?

Reply:
hmm, i'm not sure what the best fallback is tbh (I think it's pretty unlikely for us to hit the default anyway). I'll make the default `YES` to be consistent.